src/Operation/FindOneAndUpdate.php

```diff
@@ -137,11 +149,14 @@ public function __construct($databaseName, $collectionName, $filter, $update, ar
      * @see Executable::execute()
      * @param Server $server
      * @return array|object|null
-     * @throws UnsupportedException if collation or write concern is used and unsupported
+     * @throws UnsupportedException if array filters or collation or write concern is used and unsupported
      * @throws DriverRuntimeException for other driver errors (e.g. connection errors)
      */
     public function execute(Server $server)
     {
+        if (isset($this->options['arrayFilters']) && ! \MongoDB\server_supports_feature($server, self::$wireVersionForArrayFilters)) {
+            throw UnsupportedException::arrayFiltersNotSupported();
+        }
```

Comment:
If the above constructor comment is addressed, we can move this check into `FindAndModify::execute()`.